tests/mo_testing_utils.F90

```diff
@@ -0,0 +1,276 @@
+! This code is part of Radiative Transfer for Energetics (RTE)
+!
+! Contacts: Robert Pincus and Eli Mlawer
+! email:  rrtmgp@aer.com
+!
+! Copyright 2023-  Atmospheric and Environmental Research,
+!    Regents of the University of Colorado,
+!    Trustees of Columbia University in the City of New York
+! All right reserved.
+!
+! Use and duplication is permitted under the terms of the
+!    BSD 3-clause license, see http://opensource.org/licenses/BSD-3-Clause
+! ----------------------------------------------------------------------------
+module mo_testing_utils
+  use iso_fortran_env,     only: error_unit
+  use mo_rte_kind,         only: wp
+  use mo_rte_util_array,   only: zero_array
+  use mo_optical_props,    only: ty_optical_props_arry, ty_optical_props_1scl, & 
+                                 ty_optical_props_2str, ty_optical_props_nstr
+  use mo_source_functions, only: ty_source_func_lw
+  implicit none
+  private 
+  public :: allclose, ops_match, check_fluxes
+  public :: stop_on_err, report_err
+  public :: increment_with_1scl, increment_with_2str, increment_with_nstr, vr
+
+  interface allclose
+    module procedure allclose_1, allclose_2, allclose_3, allclose_4
+  end interface allclose
+
+  interface ops_match
+    module procedure ops_match_1scl, ops_match_2str, ops_match_nstr
+  end interface ops_match
+
+  interface check_fluxes
+    module procedure check_fluxes_1pair, check_fluxes_2pair
+  end interface check_fluxes
+contains
+  ! ----------------------------------------------------------------------------
+  !
+  ! Compare two arrays; return false if abs(x-y) > tol*spacing(x) for any element
+  !
+  ! ----------------------------------------------------------------------------
+  logical function allclose_1(array1, array2, tol)
+    real(wp), dimension(:), intent(in) :: array1, array2
+    real(wp), optional,     intent(in) :: tol 
+    
+    real(wp) :: tolerance 
+    if (present(tol)) then 
+      tolerance = tol 
+    else
+      tolerance = 2._wp
+    end if 
+
+    allclose_1 = all(abs(array1-array2) <= tolerance * spacing(array1))
+  end function allclose_1
+  ! ----------------------------------------------------------------------------
+  logical function allclose_2(array1, array2, tol)
+    real(wp), dimension(:,:), intent(in) :: array1, array2
+    real(wp), optional,       intent(in) :: tol 
+    
+    real(wp) :: tolerance 
+    if (present(tol)) then 
+      tolerance = tol 
+    else
+      tolerance = 2._wp
+    end if 
+
+    allclose_2= all(abs(array1-array2) <= tolerance * spacing(array1))
```

Comment:
I think that `spacing` should be calculated based on the reference data, which is often (I don't know if that's always the case) passed as the second argument.

Reply:
Made a more sweeping change in b0ca961 to clarify reference and test values and to use spacing from the ref values